core/server/proxy/src/main/java/alluxio/proxy/s3/signature/AuthorizationV4HeaderParser.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3.signature;
+
+import alluxio.proxy.s3.S3Exception;
+import alluxio.proxy.s3.S3ErrorCode;
+
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.StringTokenizer;
+
+/**
+ * Class to parse v4 auth information from header.
+ */
+public class AuthorizationV4HeaderParser implements SignatureParser {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AuthorizationV4HeaderParser.class);
+
+  private static final String CREDENTIAL = "Credential=";
+  private static final String SIGNEDHEADERS = "SignedHeaders=";
+  private static final String SIGNATURE = "Signature=";
+
+  private String mAuthHeader;
```

Comment:
final. All member variables that are not modified after constructor initialization should be final.